tests/ui_tools/test_boxes.py

```diff
@@ -356,6 +358,48 @@ def stream_id_from_name(stream_name: str) -> int:
 
         assert stream_marker == expected_marker
 
+    @pytest.mark.parametrize(['stream_name', 'stream_id', 'expected_color'], [
+      ('random stream',  1, 'general_bar'),
+      ('private stream', 2, '#c76'),
+      ('public stream',  3, '#98e'),
+     ], ids=[
+        'invalid_stream_name',
+        'valid_private_stream',
+        'valid_public_stream',
+    ])
+    def test__set_stream_write_box_style_colors(self, write_box, stream_name,
+                                                stream_id, mocker,
+                                                expected_color):
+        # FIXME: Needs refactoring?
+        stream_dict = {
+            2: {'name': 'private stream', 'invite_only': True,
+                'color': '#c76'},
+            3: {'name': 'public stream',  'invite_only': False,
+                'color': '#98e'},
+        }
+        write_box.model.stream_dict = stream_dict
+        write_box.stream_id = stream_id
+        write_box.stream_box_view(stream_id)
+
+        def is_valid_stream(stream_name: str) -> bool:
+            for stream in stream_dict.values():
+                if stream['name'] == stream_name:
+                    return True
+            return False
+
+        def stream_id_from_name(stream_name: str) -> int:
+            for stream_id, stream in stream_dict.items():
+                if stream['name'] == stream_name:
+                    return stream_id
+            raise RuntimeError("Invalid stream name.")
+
+        write_box.model.is_valid_stream = is_valid_stream
+        write_box.model.stream_id_from_name = stream_id_from_name
+        write_box._set_stream_write_box_style(stream_name)
+
+        assert (str(write_box.header_write_box[0].get_text()[1][0][0])
+                == expected_color)
```

Comment:
Actually, the `stream_marker.get_text()` object is like `('#', [('#ccc', 1)])`. So to get the color I used those indices.
Now I realized that it could be much easier if I use `get_text_obj in expected_color` :)